src/linker/Linker.Dataflow/ReflectionMarker.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using Mono.Cecil;
+using Mono.Linker.Steps;
+
+namespace Mono.Linker.Dataflow
+{
+	public readonly struct ReflectionMarker
+	{
+		readonly LinkContext _context;
+		readonly MarkStep _markStep;
+
+		public ReflectionMarker (LinkContext context, MarkStep markStep)
+		{
+			_context = context;
+			_markStep = markStep;
+		}
+
+		internal void MarkTypeForDynamicallyAccessedMembers (in MessageOrigin origin, TypeDefinition typeDefinition, DynamicallyAccessedMemberTypes requiredMemberTypes, DependencyKind dependencyKind, bool declaredOnly = false)
+		{
+			foreach (var member in typeDefinition.GetDynamicallyAccessedMembers (_context, requiredMemberTypes, declaredOnly)) {
+				switch (member) {
+				case MethodDefinition method:
+					MarkMethod (origin, method, dependencyKind);
+					break;
+				case FieldDefinition field:
+					MarkField (origin, field, dependencyKind);
+					break;
+				case TypeDefinition nestedType:
+					MarkType (origin, nestedType, dependencyKind);
+					break;
+				case PropertyDefinition property:
+					MarkProperty (origin, property, dependencyKind);
+					break;
+				case EventDefinition @event:
+					MarkEvent (origin, @event, dependencyKind);
+					break;
+				case InterfaceImplementation interfaceImplementation:
+					MarkInterfaceImplementation (origin, interfaceImplementation, dependencyKind);
+					break;
+				}
+			}
+		}
+
+		internal bool TryResolveTypeNameAndMark (string typeName, MessageOrigin origin, [NotNullWhen (true)] out TypeDefinition? type)
+		{
+			// Only difference here is whether we treat arrays specially. Maybe we should do that more frequently elsewhere too.
+			// _reflectionMarker.MarkType (origin, );
```

Comment:
nit: this comment seems kind of unfinished

Reply:
Thanks, this was accidentally left over.